src/rules/no-floating-promises.ts

```diff
@@ -0,0 +1,10 @@
+import tsPlugin from "@typescript-eslint/eslint-plugin";
+
+const baseRule = tsPlugin.rules["no-floating-promises"];
+
+if (!baseRule) throw Error("no-floating-promises is not available");
+
+baseRule.meta.messages.floatingVoid +=
+  " If you’re using `useEffect`, try some helper function like `useAsyncEffectAndCatch`.";
```

Comment:
@denis-sokolov This scopes the note to the motivating case, and is simpler than using a helper.

Reply:
This is a very smart idea. Fewer moving parts, straight to the point.

One note though is that you’re potentially mutating the original rule, which can be fine, but is generally undesirable. Could you try cloning and see if it still works?

```ts
export default {
  ...baseRule,
  meta: {
    ...baseRule.meta,
    messages: {
      ...baseRule.meta.messages,
      floatingVoid: baseRule.meta.messages.floatingVoid +
        " If you’re using `useEffect`, try some helper function like `useAsyncEffectAndCatch`.";
    }
  }
}
```